library/src/domain/Co2Eq/Co2EqUnit.ts

```diff
@@ -0,0 +1,73 @@
+import { pipe } from "fp-ts/es6/function";
+import * as Codec from "io-ts/es6/Codec";
+import { failure, success } from "io-ts/es6/Decoder";
+
+const textToUnit = (text: string) =>
+  pipe(
+    [Co2EqUnit.GRAM, Co2EqUnit.KILOGRAM, Co2EqUnit.TON].flatMap((unit) => {
+      const { singular, plural } = unit;
+      return [
+        [singular, unit],
+        [plural, unit],
+      ];
+    }),
+    Object.fromEntries
+  )[text];
+
+export class Co2EqUnit {
+  singular;
+  plural;
+  magnitude;
+
+  constructor(props: Co2EqUnitProps) {
+    this.singular = props.singular;
+    this.plural = props.plural;
+    this.magnitude = props.magnitude;
```

Comment:
I'm not sure these are used currently? And it's not clear what is the goal of `magnitude` 👀

Reply:
I'm now using these here: https://github.com/howmuchgreen/howmuchcarbon/blob/fd814d9c5ff1a9bda88e93fc490693d276ea9e5d/library/src/domain/Co2Eq/Co2Eq.ts#L19